### PR TITLE
Refactor code for embark 4.0

### PR DESCRIPTION
**Pull request by Reijnn**
I followed along with the tutorial with the new embark version. Works great!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+# [4.0.0-beta.1](https://github.com/embark-framework/embark/compare/v4.0.0-beta.0...v4.0.0-beta.1) (2019-03-18)
+
+
+### Features
+
+* add repository.directory field to package.json ([a9c5e1a](https://github.com/embark-framework/embark/commit/a9c5e1a))
+* **@embark/pipeline:** Add `enabled` property to pipeline config ([5ea4807](https://github.com/embark-framework/embark/commit/5ea4807))
+* normalize README and package.json bugs, homepage, description ([5418f16](https://github.com/embark-framework/embark/commit/5418f16))
```
Comment (0x-r4bbit) [inline CHANGELOG.md:29]: This was probably a copy and paste bug. This file doesn't belong here :)

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
```
Comment (0x-r4bbit) [inline LICENSE:22]: No need to introduce a license for upgrading the DApp to v4 :)

File: README.md [hunk 3]
```diff
@@ -1,5 +0,0 @@
-# Embark DReddit Tutorial
-
-This repository contains the code and commits for the Embark DReddit tutorial which consists of three parts.
-
-Every commit represents one step within the tutorial, making it easy to review changes.
```
Comment (0x-r4bbit) [inline README.md:5]: Please do not remove this file.

File: package.json [hunk 4]
```diff
@@ -1,17 +1,20 @@
 {
-  "name": "dreddit",
+  "name": "red",
```
Comment (0x-r4bbit) [inline package.json:2]: huh?

File: package.json [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
   "version": "0.0.1",
   "description": "",
   "scripts": {
     "test": "embark test"
   },
+  "keywords": [],
```
Comment (0x-r4bbit) [inline package.json:8]: This change is not needed

File: package.json [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
   "author": "",
-  "license": "ISC",
-  "homepage": "",
-  "devDependencies": {},
+  "license": "MIT",
+  "devDependencies": {
+    "embarkjs-connector-web3": "^4.0.0"
+  },
```
Comment (0x-r4bbit) [inline package.json:13]: This should be a `dependencies` entry instead.

File: package.json [hunk 7]
```diff
@@ -1,17 +1,20 @@
...
   "dependencies": {
     "dateformat": "^3.0.3",
     "react": "^16.4.2",
     "react-dom": "^16.4.2"
-  }
+  },
+  "gitHead": "21e23313bab22b7646df08935f565378d88a93d9"
```
Comment (0x-r4bbit) [inline package.json:19]: This is not necessary. Any reason you added this?

**Review by 0x-r4bbit — CHANGES_REQUESTED** [review]
Thank you @Reijnn for sending this PR!

Unfortunately, it's not mergable for a couple of reasons:

- There are plenty changes in here that are unrelated to upgrading the code base to Embark v4
- This repository keeps a clean commit history with steps (1 - 17) so people can follow along the tutorial. For that very reason, simply adding changes and merging them doesn't really do the trick because all previous commits would still use an older version. 

In other words, instead of merging your changes, I'll have to rebase the history to preserve the steps we need for the tutorial.

If you don't mind, simply update this PR to only introduce the changes that are really needed and I'll take care of rebasing those into the history.

Thanks!!
